### ISSUE TITLE
Failed to setup Music Assistant  {"message":"Invalid handler specified"}

### ISSUE THREAD

**Issue by madtoko** [music-assistant/hass-music-assistant#1091]
### What version of Music Assistant has the issue?

1.30.1

### The problem

Failed to setup Music Assistant on Home Assistant 2023.2.2
 {"message":"Invalid handler specified"}

### How to reproduce

Make sure that you have the [Home Assistant Community Store](https://hacs.xyz/) installed.
In the HACS panel, go to integrations.
Search for Music Assistant and click Download this repository with HACS
Restart Home Assistant.
Go to Configuration -> Integrations and click the big + button.
Look for Music Assistant and click to add it.
If Home Assistant does not show, refresh your brower (cache).
Follow the steps for initial configuration, like what players you want to use and music providers.

### Relevant log output

1)
Logger: homeassistant.config_entries
Source: config_entries.py:816
First occurred: 14:29:36 (1 occurrences)
Last logged: 14:29:36

Error occurred loading configuration flow for integration mass: cannot import name 'MusicAssistant' from partially initialized module 'music_assistant' (most likely due to a circular import) (/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/music_assistant/__init__.py)

2)
Logger: homeassistant.loader
Source: lib/python3.10/site-packages/aiofiles/os.py:31
First occurred: 14:49:05 (1 occurrences)
Last logged: 14:49:05

Unexpected exception importing platform custom_components.mass.config_flow
Traceback (most recent call last):
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/homeassistant/loader.py", line 749, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/homeassistant/loader.py", line 766, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/data/data/com.termux/files/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/data/data/com.termux/files/home/.homeassistant/custom_components/mass/__init__.py", line 22, in <module>
    from music_assistant import MusicAssistant
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/music_assistant/__init__.py", line 3, in <module>
    from .mass import MusicAssistant  # noqa
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/music_assistant/mass.py", line 19, in <module>
    from music_assistant.controllers.music import MusicController
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/music_assistant/controllers/music.py", line 31, in <module>
    from music_assistant.music_providers.filesystem import FileSystemProvider
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/music_assistant/music_providers/filesystem/__init__.py", line 3, in <module>
    from .filesystem import FileSystemProvider  # noqa
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/music_assistant/music_providers/filesystem/filesystem.py", line 14, in <module>
    from aiofiles.os import wrap
  File "/data/data/com.termux/files/home/.homeassistant/lib/python3.10/site-packages/aiofiles/os.py", line 31, in <module>
    link = wrap(os.link)
AttributeError: module 'os' has no attribute 'link'

### Additional information

Python 3.10.9 termux installation

### What version of Home Assistant Core are your running

2023.2.2

### What type of installation are you running?

Home Assistant Core

### On what type of hardware are you running?

Alternative

**Comment by marcelveldt** [comment]
Sorry, your platform is unsupported to run Music Assistant, only any of the officially supported installation types are supported. Future versions of Music Assistant will also no longer work with HA core installs. It is simply too hard to maintain with all dependencies and moving parts.

**Comment by madtoko** [comment]
Hello Marcel.
I uninstalled aiofiles 22.1.0 and installed aiofiles 0.8.0 and Music Assistant did the initial setup with no errors and is working perfectly.
Thanks :)

**Comment by madtoko** [comment]
I uninstalled aiofiles 22.1.0 and installed aiofiles 0.8.0 and Music Assistant did the initial setup with no errors and is working perfectly.

**Comment by xxxserxxx** [comment]
I have this issue as well. I do _not_ have aiofiles installed, and my version of HA is 2024.9.3, running in a container, with HACS 2.0.1. Installation through HACS went with no issue. After restart, Music Assistant is _not_ auto-detected, so I tried to add the integration manually `Settings -> Devices & Services -> Add Integration` and select Music Assistant, at which point it thinks a while and then pops up the message:

```
Config flow could not be loaded: {"message":"Invalid handler specified"}
```

I suspect that the integration is oriented to an externally running Music Assistant, rather than one installed via HACS, and this is resulting in the error; however, while Music Assistant now shows up in HACS under the installed services, I'm at a loss to figure out how to configure it. The Music Assistant Update entity shows up in Entities, but no other integration or entity is visible. Nothing shows up in the Overview view, where things are normally auto-added (I've left that alone and primarily use a different, custom, dashboard).

There's a single related error in the logs:

```
Logger: homeassistant.config_entries
Source: config_entries.py:2894
First occurred: 1:42:01 PM (4 occurrences)
Last logged: 1:55:01 PM

Error occurred loading flow for integration mass: No module named 'music_assistant'
```

The 4-count seems to correlate to the number of times I tried to manually add the integration.

1. Installed via HACS (and restarted), do I still need to manually add the integration?
2. If not, where do I go to configure this thing?

**Comment by OzGav** [comment]
@xxxserxxx Have you installed the server?

**Comment by xxxserxxx** [comment]
Nope. The installation instructions implied (or, at least, I interpreted them) as there being two ways to install Music Assistant: via HACS, in which case no separate server installation was required; or via the Integration, in which case a separate Music Assistant server would be needed. If that's _not_ the case, then I'm not clear on why there are two different processes for installing the code (HACS instructions, and Integration instructions). 

I used HACS.
